packages/utils/src/cache/distributed-cache.ts

```diff
@@ -0,0 +1,307 @@
+import { Logger } from './helpers.js';
+import { Mutex } from 'async-mutex';
+import { CoreCache } from './core-cache.js';
+import { CacheMessenger } from './messenger.js';
+import type { CacheConfig } from './core-cache.js';
+import { TimeoutTaskRunner } from '../async.js';
+import { CACHE_PREFIX, MESSENGER_NO_MESSAGE_DELAY } from './constants.js';
+
+const logger = new Logger();
+logger.timeStart(window.name);
+
+/**
+ * Mutual exclusion, prevent getting requests state concurrently from multi tabs/windows
+ */
+const mutex = new Mutex();
+
+type StorageNames = {
+  requests: string
```

Comment:
If request is singular, I would use just
```suggestion
  request: string
```
or 
```suggestion
  requested: string
```

Reply:
I mention this issue in the above discussion, What do you think about it?

Reply:
Updated